src/validate_pyproject/plugins/__init__.py

```diff
@@ -82,7 +82,7 @@ def __init__(self, tool: str, schema: Schema):
 
     @property
     def id(self) -> str:
-        return self.schema.get("id", "MISSING ID")
+        return self.schema.get("$id", "MISSING ID")
```

Comment:
> So maybe we can store the plugin name it came from too and use that.

You mean for the `sotred_plugin.id`?

If that is the idea, maybe something like this?

```python
class StoredPlugin:
    def __init__(self, tool: str, schema: Schema, fallback_id: str): ...

...

def load_from_multi_entry_point(...):
    ...
    base_id = entry_point.name
    for tool, schema in output["tools"].items():
        yield StoredPlugin(tool, schema, f"{base_id}:{tool}")
    for i, schema in enumerate(output.get("schemas", [])):
        yield StoredPlugin("", schema, f"{base_id}:{i}")
```

Or use `base_id = f"{fn.__module__}.{fn.__name__}"`

(Untested code)

Is that more or less what you had in mind in this suggestion?

Reply:
Yes, that's roughly what I was thinking. I think we use `name or id` in a few places, need to make sure none of those assume id is unique. If `id` is `$id`, then it is unique. Looks like your first suggestion would handle that, though. 👍

I was able to reproduce the sorting bug.

Reply:
I suppose that if the schema is not a tool, it is also reasonable to require `$id` to be present? (Otherwise how is the schema referenced?).